lib/apimatic-core/utilities/api_helper.rb

```diff
@@ -442,6 +442,47 @@ def self.map_response(obj, keys)
       val
     end
 
+    # Apply unboxing_function to additional properties from hash.
+    # @param [Hash] hash The hash to extract additional properties from.
+    # @param [Proc] unboxing_function The deserializer to apply to each item in the hash.
+    # @return [Hash] A hash containing the additional properties and their values.
+    def self.get_additional_properties(hash, unboxing_function, as_array: false,
+                                       as_dict: false)
+      additional_properties = {}
+
+      # Iterate over each key-value pair in the input hash+
+      hash.each do |key, value|
+        # If the value is a complex structure (Hash or Array), we apply apply_unboxing_function.
+        additional_properties[key] = if as_array || as_dict
+                                       # Call apply_unboxing_function recursively if the value is a hash or array
+                                       apply_unboxing_function(value, unboxing_function, as_array: as_array,
+                                                                                         as_dict: as_dict)
+                                     else
+                                       # Apply the unboxing function directly for simple values
+                                       unboxing_function.call(value)
+                                     end
+      rescue StandardError
+        # Optionally log the error message or handle it as needed
+        # puts "Error processing key '#{key}': #{e.message}"
+        # Ignore the exception and continue processing
+      end
+
+      additional_properties
+    end
+
+    def self.apply_unboxing_function(obj, unboxing_function, as_array: false, as_dict: false)
+      if as_dict && obj.is_a?(Hash)
+        # If `obj` is a Hash and `as_dict` is true, apply the unboxing function to each value in the hash.
+        obj.transform_values { |v| apply_unboxing_function(v, unboxing_function, as_dict: true) }
+      elsif as_array && obj.is_a?(Array)
+        # If `obj` is an Array, apply the unboxing function to each element.
+        obj.map { |element| unboxing_function.call(element) }
+      else
+        # Otherwise, apply the unboxing function directly to the object.
+        unboxing_function.call(obj)
+      end
+    end
```

Comment:
This function lacks the support for array of map and map of array because in case of hash, it is passing the as_dict where as it could be an array or a simple type. In case of array, the unboxing funtion is called directly to the elements but it could be a map. We need to support the array of maps and map of array cases as well.

Reply:
I have visited the python implementation and have made the modifications here